src/v/config/property.h

```diff
@@ -130,8 +166,102 @@ class property : public base_property {
     constexpr static auto noop_validator = [](const auto&) {
         return std::nullopt;
     };
+
+    boost::intrusive::
+      list<binding<T>, boost::intrusive::constant_time_size<false>>
+        _bindings;
+    friend class binding<T>;
 };
 
+/**
+ * A property binding contains a copy of the property's value, which
+ * will be updated in-place whenever the property is updated in the
+ * cluster configuration.
+ *
+ * This is useful for classes that want a copy of a property without
+ * having to write their own logic for subscribing to value changes.
+ */
+using intrusive_auto_unlink = boost::intrusive::list_base_hook<
+  boost::intrusive::link_mode<boost::intrusive::auto_unlink>>;
```

Comment:
we have some pre-defined types in `src/v/utils/intrusive_list_helpers.h` if this particular flavor isn't covered it might be worth giving it a name and sticking in the utils header

Reply:
Ah-ha!

Our existing definitions were for _member_ rather than _base_ versions of the hooks, but that's fine, I like them just as well.  Converted this to use those.